src/Model/ProfileModel.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace Klaviyo\Model;
+
+use Klaviyo\Exception\KlaviyoException;
+
+/**
+ * Class defining a profile in Klaviyo with methods to set
+ * various special properties.
+ */
+class ProfileModel extends BaseModel
+{
+    public $id;
+    public $email;
+    public $firstName;
+    public $lastName;
+    public $phoneNumber;
+    public $title;
+    public $city;
+    public $organization;
+    public $region;
+    public $country;
+    public $zip;
+    protected $customAttributes;
+
+    public static $specialAttributes = [
+        '$id',
+        '$email',
+        '$first_name',
+        '$last_name',
+        '$organization',
+        '$title',
+        '$address1',
+        '$address2',
+        '$city',
+        '$region',
+        '$zip',
+        '$country',
+        '$timezone',
+        '$phone_number',
+        '$ios_tokens',
+    ];
+
+    public static $identifyAttributes = [
+        '$email',
+        '$id',
+        '$phone_number',
+        '$ios_tokens',
+    ];
+
+    public function __construct( array $configuration ) {
+        if (empty(array_intersect_key( $configuration, array_flip(self::$identifyAttributes) ))) {
+            throw new KlaviyoException(
+                sprintf(
+                    'ProfileModel requires one of the following fields for identification: %s',
+                    implode(', ', self::$identifyAttributes)
+                )
+            );
+        }
+        $this->setAttributes( $configuration );
+    }
+
+    protected function setAttributes( array $configuration ) {
+        foreach ( $configuration as $key => $value ) {
+            if ( $this->isSpecialAttribute($key) ) {
+                $this->{ltrim($key, '$')} = $value;
+            }
+            
+        }
+
+        $this->setCustomAttributes( $configuration );
+    }
+
+    private function setCustomAttributes( array $configuration ) {
+        $customAttributeKeys = array_flip(
+            array_filter(
+                array_keys( $configuration ),
+                'self::isCustomAttribute'
```

Comment:
should this be 
```suggestion
                self::isCustomAttribute
```

Reply:
This is ok, array_filter is weird. https://www.php.net/manual/en/function.array-filter.php